src/Filter/UuidFilter.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace TraderInteractive\Filter;
+
+use InvalidArgumentException;
+use TraderInteractive\Exceptions\FilterException;
+
+final class UuidFilter
+{
+    /**
+     * @var string
+     */
+    const FILTER_ALIAS = 'uuid';
+
+    /**
+     * @var string
+     */
+    const UUID_PATTERN_FORMAT = '^[0-9A-F]{8}-[0-9A-F]{4}-[%d][0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$';
+
+    /**
+     * @var string
+     */
+    const FILTER_ERROR_FORMAT = "Value '%s' is not a valid UUID. Versions checked (%s)";
+
+    /**
+     * @var array
+     * @internal
+     */
+    const VALID_UUID_VERSIONS = [1,2,3,4,5,6,7];
+
+
+    /**
+     * Filters a given string values to a valid UUID
+     *
+     * @param string|null $value     The value to be filtered.
+     * @param bool        $allowNull Flag to allow value to be null.
+     * @param array       $versions  List of specific UUID version to validate against.
+     *
+     * @return string|null
+     *
+     * @throws FilterException Thrown if value cannot be filtered as an UUID.
+     */
+    public static function filter(
+        string $value = null,
+        bool $allowNull = false,
+        array $versions = self::VALID_UUID_VERSIONS
+    ) {
+        if (self::valueIsNullAndValid($allowNull, $value)) {
+            return null;
+        }
+
+        self::validateVersions($versions);
+        foreach ($versions as $version) {
+            $pattern = sprintf(self::UUID_PATTERN_FORMAT, $version);
+            if (preg_match("/{$pattern}/i", $value)) {
+                return $value;
+            }
+        }
```

Comment:
Should this support [Nil UUIDs](https://datatracker.ietf.org/doc/html/rfc4122#section-4.1.7)? The `[%d]` and `[89AB]` constraints in the pattern currently prevent them from being used.

Reply:
That's a thing?! It's a good question. Would the allow null option control if this was considered valid?

Reply:
I've updated the filter to allow nil UUIDs. I'm not sure if i like the implementation, but i'd like to get feedback

Reply:
The updates look like an appropriate solution. Not sure I have any suggestion to improve the implementation.

Reply:
It's the best thing i could come up with. I'll add some examples to the README

Reply:
if approved i will also make the changes to the v3 PR

Reply:
The updated code looks good to me. Handling `null` values and nil UUIDs separately gives more flexibility to the filter than combining them together in some way.